cpp/benchmarks/join/join_benchmark.cu

```diff
@@ -47,10 +47,31 @@ static void BM_join(benchmark::State &state)
 
   // Generate build and probe tables
 
-  auto build_key_column =
-    cudf::make_numeric_column(cudf::data_type(cudf::type_to_id<key_type>()), build_table_size);
-  auto probe_key_column =
-    cudf::make_numeric_column(cudf::data_type(cudf::type_to_id<key_type>()), probe_table_size);
+  auto build_random_null_mask = [](int size) {
+    if (Nullable) {
+      // roughly 25% nulls
+      auto validity = thrust::make_transform_iterator(
+        thrust::make_counting_iterator(0), [](auto i) { return (std::rand() & 3) == 0; });
```

Comment:
Super super minor point, but it would be mildly better to srand() with a known fixed seed somewhere in this module so that we're truly reproducing the same thing every time.